spec/features/trainees/recommending_for_qts_spec.rb

```diff
@@ -5,7 +5,8 @@
 feature "Recommending for QTS", type: :feature do
   before do
     given_i_am_authenticated
-    given_a_trainee_exists
+    given_a_trainee_exists(:with_placement_assignment)
+    stub_dttp_placement_assignment_request(outcome_date: Time.zone.today, status: 204)
```

Comment:
Putting these 'steps' in is starting to take us back away from the easy to read behavioural style. I wasn't really sure about moving the `given_i` etc into before blocks in the first place but this is definitely starting to look wrong now. Can we get the stubbing out into somewhere else? Or leave it in the `before` which is probably the right place for it and move the steps into the scenarios. It doesn't matter if they are duplication as it's more important to read the scenario and understand it without jumping around which was the problem we were solving using this style in the first place.

Reply:
Fair point. We need to call `stub_dttp_placement_assignment_request` after the trainee is created, so I've created a new step `and_a_trainee_exists_ready_for_qts` which looks like this: 

```ruby
def and_a_trainee_exists_ready_for_qts
  given_a_trainee_exists(:with_placement_assignment)
  stub_dttp_placement_assignment_request(outcome_date: Time.zone.today, status: 204)
end
```

What do you think?